test/e2e/operator_test.go

```diff
@@ -338,17 +426,61 @@ func TestExternalDNSRecordLifecycle(t *testing.T) {
 	t.Fatalf("All nameservers failed to verify that DNS has been correctly set.")
 }
 
-func customResolver(nameserver string) *net.Resolver {
-	return &net.Resolver{
-		PreferGo: true,
-		Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
-			d := net.Dialer{
-				Timeout: dialTimeout,
+func verifyOpenShiftRouteSource(t *testing.T, canonicalName, host string) {
+	//var canonicalName = "router-external-dns.external-dns.example-naga.info"
+	// try all nameservers and fail only if all failed
+	recordExist := false
+	for _, nameSrv := range nameServers {
+		t.Logf("Looking for cname record in nameserver: %s", nameSrv)
+		if err := wait.PollImmediate(dnsPollingInterval, dnsPollingTimeout, func() (done bool, err error) {
+			cnames, err := lookupCNAME(host, nameSrv)
+			if err != nil {
+				t.Logf("cname lookup failed for nameserver : %s , error : %v", nameSrv, err)
```

Comment:
```suggestion
				t.Logf("CNAME lookup failed, error : %v", err)
```
As the nameserver was already mentioned at the beginning of the polling.